### ISSUE TITLE
feature request: tolerance for locate on screen

### ISSUE THREAD

**Issue by jeremygray** [asweigart/pyautogui#150]
(Picking up from the tail end of discussion on #4)

A tolerance for locating on screen would be of interest to several people.

Possible easy solution: expose the "confidence" parameter, allowing people to locate with reduced confidence ( = more tolerance for mismatches)

**Comment by bjanzen** [comment]
Add confidence value, and confirm you can use cv2:  
pyautogui.locateCenterOnScreen('foo.png', confidence=0.999)
lower default values.  I just tried it with the image not on screen, it found something similar at confidence=0.5.

**Comment by xorinus** [comment]
Is "confidence" available for the locate functions?
I get "_locateAll_python() got an unexpected keyword argument 'confidence'" when I try to use it in my scripts.

**Comment by jeremygray** [comment]
I think that if this were documented, we could close this issue. If you have opencv installed, the code works, passing `confidence=.9` (or whatever).

**Comment by RAB420** [comment]
I have installed opencv via: 
sudo python3 -m pip install opencv-contrib-python
which gave me version:  "opencv_contrib_python-4.0.0.21-cp35-cp35m-manylinux1_x86_64"
and pyautogui won't seem to recognize this.  

If someone could assist me by telling me if a particular version of opencv is required, or if I have done something incorrect you can notice?  Thank you.

Additional Notes: 
Trying to use:
`if pyautogui.locateCenterOnScreen(myVar, confidence=.9) is not None:`
but get error: TypeError: _locateAll_python() got an unexpected keyword argument 'confidence'
which as I can conclude from reading research means pyautogui is not detecting my opencv installation.

**Comment by Sharbell** [comment]
You need to install opencv-python
`pip install opencv-python`

**Comment by kukai10** [comment]
So i have opencv installed and the locate function is working.  but how does the confidence work?  does it work for rescaled images? what about color, shape, size, flipped/inverted, etc?

**Comment by jeremygray** [comment]
`confidence` is the value that gets used when thresholding the normalized cross-correlation values. It will not handle changes in image size or orientation at all gracefully, although tiny changes could still match. Not sure what happens with color but you might be able to do locate() using grayscale images.

**Comment by waterfronter** [comment]
I install cv2, so calling pyautogui.locateOnScreen will eventually call _locateAll_opencv in pyscreeze moudle.
 First，I called pyautogui.locateOnScreen with param confidence=0.8, and found this value doesn't work, still return None.
So I realized pyautogui didn't transfer the param confidence to pyscreeze. Therefore, I changed confidence to 0.8 (default value is 0.999) in function _locateAll_opencv directly, and it worked!

Here I suggest pyautogui use confidence param in locate related functions.
